### ISSUE TITLE
Shortcut for reducing counters not working properly

### ISSUE THREAD

**Issue by spacian** [Cockatrice/Cockatrice#3049]
<b>OS: Windows 8.1</b>

<b>Cockatrice version: Version 2.4.0 (2017-11-19)</b>

__________

So when I assign a shortcut for reducing the [red/green/yellow] counters on a permanent, this only works for all the permanents that are currently in play while I'm making the change (so a game is already open).

When I put the next permanent in play and give it some counters, I can't reduce the counters again on that permanent with the given shortcut. It still works for all the permanents that were in play at the time of the shortcut definition though. The issue is the same when I start a new game with the shortcuts already defined. I can't reduce the number of counters.

So for example:
I start a new local game with 1 player, load a deck with 60 plains. I mulligan and play Plains1. I add 2 red counters to it. Now I define the shortcut for reducing red counters, e.g. 'Ctrl+1'. I can now select Plains1 and remove the counter again with the shortcut. Now I put Plains2 into play and add a red counter to it. After selecting it, I try to remove a red counter with the defined shortcut. It doesn't work. Now I select Plains1 again and try to remove the 2nd counter with the shortcut. It works.

Hope I was able to give some insight in that bug :)

Best regards,
spacian

**Comment by ZeldaZach** [comment]
Local games are always buggy with stuff like this... Honestly I don't know why we keep supporting them as they just complicate the code :(

**Comment by skwerlman** [comment]
what if we redid local games so they were actually run on a local server, by basically spawning a servatrice instance and connecting to it?

**Comment by spacian** [comment]
To clarify: This is also an issue on server games (just tested it).

**Comment by Daenyth** [comment]
@skwerlman I've floated that idea before. It's a fair bit of work unfortunately. On the flip side, as we get the codebase cleaner over time, local mode should actually get more robust. The problem is that it's kind of special cased in the code instead of just looking like a normal server connection.

@ZeldaZach I've talked about removing it entirely before and the feedback from the user base is really negative - people use it a lot
